src/lib.rs

```diff
@@ -2682,22 +2688,22 @@ pub unsafe trait FromBytes: FromZeros {
     #[must_use = "has no side effects"]
     #[inline]
     fn mut_from_suffix(
-        bytes: &mut [u8],
+        source: &mut [u8],
     ) -> Result<(&mut [u8], &mut Self), CastError<&mut [u8], Self>>
     where
         Self: IntoBytes + KnownLayout,
     {
         util::assert_dst_is_not_zst::<Self>();
-        mut_from_prefix_suffix(bytes, None, CastType::Suffix).map(swap)
+        mut_from_prefix_suffix(source, None, CastType::Suffix).map(swap)
     }
 
-    /// Interprets the given `bytes` as a `&Self` with a DST length equal to
+    /// Interprets the given bytes as a `&Self` with a DST length equal to
     /// `count`.
     ///
-    /// This method verifies that `bytes.len() == size_of::<T>() * count` and
-    /// that `bytes` is aligned to `align_of::<T>()`. It also ensures that
-    /// `sizeof::<T>() * count` does not overflow a `usize`. If any of the
-    /// length, alignment, or overflow checks fail, it returns `Err`.
+    /// This method attempts to return a reference to `source` interpreted as a
+    /// `Self` with `count` trailing elements. If the length of `source` is not
+    /// a valid size of `Self`, or if `source` is not appropriately aligned,
```

Comment:
"Valid size of `Self`" isn't the right condition when an explicit count is given.

Reply:
[Done](https://github.com/google/zerocopy/compare/f31d47afecf6f6652a2343d33ea5bc13662b1800..0c1f3c5d10b2842e836b549d7a8fbea1d5613700#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R2705)